guides/TERMINOLOGY.md

```diff
@@ -0,0 +1,7 @@
+#Terminology
+- Product - A piece of software controlled by an operator (e.g. RHSSO, 3scale, etc)
+- Resource - Any CRD that is managed by the product's operator. 
+- Reconciler - this refers to the package in the integreatly-operator that is concerned with reconciling a single 
+products operator, resources and configuration.
+- Stage - A set of products that can be installed at the same time as each other.
+- Installation Type - Defines a set of stages to process.
```

Comment:
From this definition it might not be obvious that installation type also defines set of products that will be installed.

Reply:
updated this.